stories/utils/generate-widget-stories.js

```diff
@@ -82,6 +95,7 @@ export function generateReportBasedWidgetStories( {
 	group,
 	data,
 	options,
+	additionalVariants = {},
 	additionalVariantCallbacks = {},
```

Comment:
The behavior here feels a bit confusing to me, maybe only because these two are named so similarly but do actually unrelated things. `additionalVariants` adds actual additional variants while `additionalVariantCallbacks` expands already existing variants with extra data. Furthermore I think it's odd that they don't support each other, e.g. I currently cannot provide an `additionalVariantsCallback` for one of the `additionalVariants`.

I suggest the following:
* Make `additionalVariants` a map of `variantName => variantFixtureData` pairs where `variantFixtureData` is an object that has `data` and `options`. The utility should then fill this data the same way as for the automatic existing variants.
* And then, add support for `additionalVariants` to `additionalVariantCallbacks` so that you can also provide extra logic for an additional variant in an additional variant callback.

Reply:
@ryanwelcher I think the change you added here now makes it more complicated than it was before - why change `additionalVariants` to be an array? This makes the code unnecessarily complex. The format of `additionalVariantCallbacks` shouldn't have been changed, and `additionalVariants` should be a map (i.e. object) so that they can be easily looked up by name. Having the name as first value of an array is counter-intuitive I'd say, plus it requires iterating through an array in several places which can be avoided by using an object.